src/prpy/perception/chisel.py

```diff
@@ -0,0 +1,139 @@
+import logging
+from base import PerceptionModule, PerceptionMethod
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+class ChiselModule(PerceptionModule):
+    
+    def __init__(self, env, mesh_client=None, service_namespace=None,
+               detection_frame=None, destination_frame=None, reference_link=None):
+        """
+        @param env The OpenRAVE environment
+        @param mesh_client An instance of the Chisel sensor system (if None, one is created)
+        @param service_namespace The namespace to use for Chisel service calls (default: Chisel)
+        @param destination_frame The tf frame the detected chisel kinbody should be transformed to
+        (default map)
+        @param reference_link A link on an OpenRAVE kinbody that corresponds to the tf frame
+        specified by the destination_frame parameter. If not provided it is assumed that the 
+        transform from the destination_frame tf frame to the OpenRAVE environment is the identity
+        """
+        import rospy
+
+        try:
+            rospy.init_node('chisel_detector',anonymous=True)
+        except rospy.exceptions.ROSException:
+            pass
+
+        if service_namespace is None:
+            service_namespace = 'Chisel'
+
+        if mesh_client is None:
+            import openravepy
```

Comment:
Make this a file-level import. It's not like this script would be useful without `openravepy` installed. :smile: